presto-main/src/main/java/com/facebook/presto/operator/ArrayUnnester.java

```diff
@@ -58,13 +45,15 @@ public boolean hasNext()
     @Override
     public final int getChannelCount()
     {
-        return channelCount;
+        return 1;
     }
 
     @Override
     public final void appendNext(PageBuilder pageBuilder, int outputChannelOffset)
     {
-        appendTo(pageBuilder, outputChannelOffset);
+        BlockBuilder blockBuilder = pageBuilder.getBlockBuilder(outputChannelOffset);
+        elementType.appendTo(arrayBlock, position, blockBuilder);
```

Comment:
`requireNonNull(arrayBlock`? Or `checkState`?

Reply:
This is interesting, because in the `setBlock()` method below `arrayBlock` is `@Nullable`. But, when it's null `appendTo` call may fail with NPE (which I have never seen). Will take a detailed look.

Reply:
OK it was fine before, because `appendNext` was called after checking `hasNext` and `hasNext` will return false when the `arrayBlock` is null. I will add a `checkState`.